docs/how-to-guides/tracing/trace_span.md

```diff
@@ -0,0 +1,152 @@
+
+**Flow**
+
+The Flow span encapsulates the execution details of a flow within PromptFlow.
+
+| Attribute | Type | Description | Examples | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|---|---|---|---|---|
+| span_type | string | Designates the span as a Flow type. | Flow | `Required` |
+
+**Embedding**
+
+The Embedding span is dedicated to recording the details of embedding calls within PromptFlow.
+
+| Attribute | Type | Description | Examples | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|---|---|---|---|---|
+| span_type | string | Denotes the span as an Embedding type. | Embedding | `Required` |
+| llm.usage.total_tokens | int | Total number of tokens used, sum of prompt and response tokens. | `180` | `Required` |
+| llm.usage.prompt_tokens | int | Number of tokens used in the prompt for the embedding call. | `100` | `Required` |
+| llm.usage.completion_tokens | int | Number of tokens used in the response from the embedding call. | `80` | `Required` |
+| llm.response.model | string | Identifies the LLM model used for generating the embedding. | text-embedding-ada-002 | `Required` |
+
+| Event | Payload Description | Payload Examples | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|---|---|---|---|
+| promptflow.embedding.embeddings | Details the embeddings generated by a call. | ```[{"embedding.vector":"","embedding.text":"When does a pipeline job reuse a previous job's results in Azure Machine Learning?"}]``` | `Required` |
+
+**Retrieval**
+
+The Retrieval span type is specifically designed to encapsulate the execution details of a retrieval task within PromptFlow.
+
+| Attribute | Type | Description | Examples | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|---|---|---|---|---|
+| span_type | string | Labels the span as a Retrieval type. | Retrieval | `Required` |
+
+| Event | Payload Description | Payload Examples | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|---|---|---|---|
+| promptflow.retrieval.query | Captures the text of the retrieval query. | ```"When does a pipeline job reuse a previous job's results in Azure Machine Learning?"``` | `Required` |
+| promptflow.retrieval.documents | Details the list of retrieved documents relevant to the query. | ```[{"document.id":"https://componentsdk.azurewebsites.net/howto/caching-reuse.html","document.score":2.677619457244873,"document.content":"# Component concepts &..."},{"document.id":"https://learn.microsoft.com/en-us/azure/machine-learning/v1/reference-pipeline-yaml","document.score":2.563112735748291,"document.content":"etc. \|\r\n\| runconfig \| T..."}]``` | `Required` |
+
+# Conclusion
+
```

Comment:
do we support the customized span kind? any doc or guidance on that?

Reply:
Current, we do not have any customized span kind.